community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/legacy/MultipleBackupDeletionPolicy.java

```diff
@@ -19,19 +19,14 @@
  */
 package org.neo4j.index.impl.lucene.legacy;
 
-import java.io.IOException;
-
 import org.apache.lucene.index.IndexCommit;
 import org.apache.lucene.index.KeepOnlyLastCommitDeletionPolicy;
 import org.apache.lucene.index.SnapshotDeletionPolicy;
 
+import java.io.IOException;
+
 public class MultipleBackupDeletionPolicy extends SnapshotDeletionPolicy
 {
-    static final String SNAPSHOT_ID = "backup";
-
-    private IndexCommit snapshot;
-    private int snapshotUsers;
-
     public MultipleBackupDeletionPolicy()
     {
         super( new KeepOnlyLastCommitDeletionPolicy() );
```

Comment:
Can we be sure that the initial snapshot keeps the files it lists, even if there are more commits? The class name looks like it will not be the case. I don't think your tests explicitly tests that because file deletion perhaps may not happen with the given load that test has.